src/Disruptor/Util/InternalUtil.cs

```diff
@@ -58,10 +58,13 @@ public static T Read<T>(object array, int index)
         Ldloc_0(); // load the object pointer as a byref
 
         Ldarg(nameof(index));
+        Conv_U(); // zero extend
+
         Sizeof(typeof(object));
         Mul(); // index x sizeof(object)
 
         Call(MethodRef.PropertyGet(typeof(InternalUtil), nameof(ArrayDataOffset)));
```

Comment:
We could avoid zero extend at all by storing `ArrayDataOffset` as `nint`

Reply:
It would make the code easier to understand as well and is exactly what `nint` was intended for